pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java

```diff
@@ -245,6 +252,12 @@ public boolean index(GenericRow row) {
     minTimeVal = Math.min(minTimeVal, timeValue);
     maxTimeVal = Math.max(maxTimeVal, timeValue);
 
+    try {
+      long msgEventTimeLagInSeconds = System.currentTimeMillis() / 1000 - (long) toSecondsTimeConverter.convert(timeValueObj);
+      serverMetrics.addTimeredTableValue(tableAndStreamName, ServerTimer.CURRENT_MSG_EVENT_TIMESTAMP_LAG, msgEventTimeLagInSeconds, TimeUnit.SECONDS);
```

Comment:
Can we split this into two pr
On Apr 18, 2016 3:10 PM, "Xiang Fu" <notifications@github.com> wrote:

> In
> pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java
> <https://github.com/linkedin/pinot/pull/107#discussion_r60143355>:
>
> > @@ -245,6 +252,12 @@ public boolean index(GenericRow row) {
> >      minTimeVal = Math.min(minTimeVal, timeValue);
> >      maxTimeVal = Math.max(maxTimeVal, timeValue);
> >
> > +    try {
> > +      long msgEventTimeLagInSeconds = System.currentTimeMillis() / 1000 - (long) toSecondsTimeConverter.convert(timeValueObj);
> > +      serverMetrics.addTimeredTableValue(tableAndStreamName, ServerTimer.CURRENT_MSG_EVENT_TIMESTAMP_LAG, msgEventTimeLagInSeconds, TimeUnit.SECONDS);
>
> Good point, I will adjust msgEventTimeLagInSeconds based on
> incomingTimeGranularity.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/linkedin/pinot/pull/107/files/a68397ebb682710975bbd488f90989a36ad724e9#r60143355>
>